sanic/request.py

```diff
@@ -53,6 +56,38 @@
     from json import loads as json_loads  # type: ignore
 
 
+@dataclasses.dataclass()
+class Credentials:
+    _auth_type: Optional[str]
+    token: Optional[str]
```

Comment:
Why `Optional`?

Reply:
Optional because there can be a None. Look at the [test-cases](https://github.com/SerGeRybakov/sanic/blob/credentials/tests/test_requests.py#L455-L626) - they are all the same as for the [token property](https://github.com/SerGeRybakov/sanic/blob/credentials/tests/test_requests.py#L455-L626). I believe they must be the same for tokens as we make a combined property.

Reply:
I am not sure I am seeing which use case you are talking about. If there is no `authorization` header, there is nothing to parse and we should not even have a `Credentials` object.

Reply:
right the first one and the last one